addons/account/tests/common.py

```diff
@@ -666,3 +666,188 @@ def get_xml_tree_from_string(self, xml_tree_str):
+        return {
+            'expected_price_unit': expected_price_unit,
+            'params': {
+                'test': 'adapt_price_unit_to_another_taxes',
+                'original_tax_values_list': original_taxes._convert_to_dict_for_taxes_computation(),
+                'new_tax_values_list': new_taxes._convert_to_dict_for_taxes_computation(),
+                'price_unit': price_unit,
+            },
+        }
+
+    def _add_test_py_results(self, test):
+        params = test['params']
+        if params['test'] == 'taxes_computation':
+            evaluation_context = self.env['account.tax']._eval_taxes_computation_prepare_context(
+                params['price_unit'],
+                params['quantity'],
+                **params['evaluation_context_kwargs'],
+            )
+            taxes_computation = self.env['account.tax']._prepare_taxes_computation(params['tax_values_list'], **params['compute_kwargs'])
+            test['py_results'] = {
+                'results': self.env['account.tax']._eval_taxes_computation(taxes_computation, evaluation_context),
+            }
+
+            if params['is_round_globally']:
+                evaluation_context = self.env['account.tax']._eval_taxes_computation_prepare_context(
+                    test['py_results']['results']['total_excluded'] / params['quantity'],
+                    params['quantity'],
+                    **params['evaluation_context_kwargs'],
+                    reverse=True,
+                )
+                test['py_results']['reverse_results'] = self.env['account.tax']._eval_taxes_computation(taxes_computation, evaluation_context)
+        elif params['test'] == 'adapt_price_unit_to_another_taxes':
+            test['py_results'] = self.env['account.tax']._adapt_price_unit_to_another_taxes(
+                params['price_unit'],
+                params['original_tax_values_list'],
+                params['new_tax_values_list'],
+            )
+        else:
+            assert False, f"Unknown tax test method: {params['test']}"
+
+    def _add_tests_py_results(self, tests):
+        for test in tests:
+            self._add_test_py_results(test)
+
+    def _add_tests_js_results(self, tests):
+        self.env['ir.config_parameter'].set_param(
+            'account.tests_shared_js_python',
+            json.dumps([test['params'] for test in tests]),
+        )
+
+        self.start_tour('/account/init_tests_shared_js_python', 'tests_shared_js_python', login=self.env.user.login)
+        results = json.loads(self.env['ir.config_parameter'].get_param('account.tests_shared_js_python', '[]'))
+        self.assertEqual(len(results), len(tests))
+
+        for test, result in zip(tests, results):
+            test['js_results'] = result
+
+    def _assert_sub_test(self, test, results):
+        def compare_taxes_computation_values(results, expected_values, rounding):
```

Comment:
Since you already define `_prepare_taxes_computation_test` as a class method, wouldn't it make sense to do it as well for this one?
Maybe even the whole content of the `if params['test'] == 'taxes_computation':` block?